src/trie.py

```diff
@@ -46,3 +46,25 @@ def contains(self, token):
             else:
                 return False
         return True
+
+    def traversal(self, start):
+        """
+        Perform a full depth-first traversal of the trie beginning at start.
+        Return a generator containing all tokens in the trie.
+        """
+        def create_token(token, current_dict):
+            for key in current_dict:
+                if key == '$':
+                    yield token
+                else:
+                    token1 = token + key
+                    current_dict1 = current_dict[key]
+                    result = create_token(token1, current_dict1)
+                    for x in result:
+                        yield x
+        token = start
+        if start in self.head.keys():
```

Comment:
- If you nonetheless prefer to use `in` operator instead of try except loop, you shouldn't use `your_dict.keys()` to iterate over a dictionary
```python
# Bad
for key in your_dict.keys():
    ...

# Better
for key in your_dict:
    ...
```